### PR TITLE
Get &attributes working properly

### LINKED ISSUE #46: '&attribtues()' is dropped at compile
jade:
```jade
div.useful&attributes(this.props.attributes)
div&attributes(this.props.attributes)
div(style={cursor: "pointer"})&attributes(this.props.attributes)
```

compiled js:
```js
  template = (function (React) {
  var fn = function(locals) {
    var tags = [];
    tags.push(React.createElement("div", {
      className: "useful"
    }));
    tags.push(React.createElement("div", {}));
    tags.push(React.createElement("div", {
      style: {
        cursor: "pointer"
      }
    }));
    if (tags.length === 1 && !Array.isArray(tags[0])) {
      return tags.pop();
    }
    tags.unshift("div", null);
    return React.createElement.apply(React, tags);
  };
  
  fn.locals = function setLocals(locals) {
    var render = this;
    function newRender(additionalLocals) {
      var newLocals = {};
      for (var key in locals) {
        newLocals[key] = locals[key];
      }
      if (additionalLocals) {
        for (var key in additionalLocals) {
          newLocals[key] = additionalLocals[key];
        }
      }
      return render.call(this, newLocals);
    }
    newRender.locals = setLocals;
    return newRender;
  };;
  return fn;
}(typeof React !== "undefined" ? React : require("../../../node_modules/react/react.js")));
```

### PR DESCRIPTION
**Pull request by ForbesLindesay**
Also updates the deps and fixes a few other unrelated issues (I should
probably have done lots of separate pull requests but oh well)
[closes #47][closes #46]

### CODE REVIEW

**Comment by gvilarino** [comment]
:+1:

**Comment by uneco** [comment]
:+1:

**Comment by hemanth** [comment]
:+1:

File: lib/parse.js [hunk 1]
```diff
@@ -11,7 +11,10 @@ var JavaScriptCompressor = require('./utils/java-script-compressor.js');
 
 var jade_join_classes = fs.readFileSync(__dirname + '/utils/jade-join-classes.js', 'utf8');
 var jade_fix_style = fs.readFileSync(__dirname + '/utils/jade-fix-style.js', 'utf8');
+var jade_fix_attrs = fs.readFileSync(__dirname + '/utils/jade-fix-attrs.js', 'utf8');
+var jade_merge = fs.readFileSync(__dirname + '/utils/jade-merge.js', 'utf8');
 var setLocals = fs.readFileSync(__dirname + '/utils/set-locals.js', 'utf8');
+var jade_fix_attrs = fs.readFileSync(__dirname + '/utils/jade-fix-attrs.js', 'utf8');
```
Comment (ForbesLindesay) [inline lib/parse.js]: This looks like it's been duplicated.

**Comment by ForbesLindesay** [comment]
I totally just did a code review of this code without realising I had written it.

File: lib/parse.js (hunk 1 above)
Comment (hemanth) [inline lib/parse.js]: Yikes! :cactus:

**Pull request merged**